src/compiler/transformers/declarations.ts

```diff
@@ -308,46 +310,93 @@ export function transformDeclarations(context: TransformationContext) {
     let exportedModulesFromDeclarationEmit: Symbol[] | undefined;
 
     const { factory } = context;
-    const host = context.getEmitHost();
-    const symbolTracker: SymbolTracker = {
-        trackSymbol,
-        reportInaccessibleThisError,
-        reportInaccessibleUniqueSymbolError,
-        reportCyclicStructureError,
-        reportPrivateInBaseOfClassExpression,
-        reportLikelyUnsafeImportRequiredError,
-        reportTruncationError,
-        moduleResolverHost: host,
-        trackReferencedAmbientModule,
-        trackExternalModuleSymbolOfImportTypeNode,
-        reportNonlocalAugmentation,
-        reportNonSerializableProperty,
-    };
+
     let errorNameNode: DeclarationName | undefined;
     let errorFallbackNode: Declaration | undefined;
 
     let currentSourceFile: SourceFile;
     let refs: Map<NodeId, SourceFile>;
     let libs: Map<string, boolean>;
     let emittedImports: readonly AnyImportSyntax[] | undefined; // must be declared in container so it can be `undefined` while transformer's first pass
-    const resolver = context.getEmitResolver();
-    const { resolver: localInferenceResolver, isolatedDeclarations } = createLocalInferenceResolver({
-        ensureParameter,
-        context,
-        visitDeclarationSubtree,
-        setEnclosingDeclarations(node) {
-            const oldNode = enclosingDeclaration;
-            enclosingDeclaration = node;
-            return oldNode;
-        },
-        checkEntityNameVisibility(name, container) {
-            return checkEntityNameVisibility(name, container ?? enclosingDeclaration);
-        },
-    });
+    const { localInferenceResolver, isolatedDeclarations, host, resolver, symbolTracker } = createTransformerServices();
     const options = context.getCompilerOptions();
     const { noResolve, stripInternal } = options;
     return transformRoot;
 
+    function createTransformerServices(): {
+        isolatedDeclarations: true;
+        resolver: IsolatedEmitResolver;
+        localInferenceResolver: LocalInferenceResolver;
+        host: undefined;
+        symbolTracker: SymbolTracker;
+    } | {
+        isolatedDeclarations: false;
+        resolver: EmitResolver;
+        localInferenceResolver: undefined;
+        host: EmitHost;
+        symbolTracker: SymbolTracker;
+    } {
+        const { isolatedDeclarations, resolver: localInferenceResolver } = createLocalInferenceResolver({
+            ensureParameter,
+            context,
+            visitDeclarationSubtree,
+            setEnclosingDeclarations(node) {
+                const oldNode = enclosingDeclaration;
+                enclosingDeclaration = node;
+                return oldNode;
+            },
+            checkEntityNameVisibility(name, container) {
+                return checkEntityNameVisibility(name, container ?? enclosingDeclaration);
+            },
+        });
+        if (isolatedDeclarations) {
+            const symbolTracker: SymbolTracker = {

```

Comment:
Can you move this out of the if statement and assign the specific case to make it more readable the difference?
i.e `symbolTracker.moduleResolverHost = host;`

Reply:
I changed a bit to move the symbol tracker creation to a single location. And not create it in isolated declaration mode.